processor/processor.go

```diff
@@ -2160,23 +2190,64 @@ func (proc *HandleT) pipelineWithPause(ctx context.Context, fn func(ctx context.
 	}
 }
 
+//`jobSplitter` func Splits the read Jobs into sub-batches after reading from DB to process.
+//`subJobMerger` func merges the split jobs into a single batch before writing to DB.
+//So, to keep track of sub-batch we have `hasMore` variable.
+//each sub-batch has `hasMore`. If, a sub-batch is the last one from the batch it's marked as `false`, else `true`.
+type subJobT struct {
+	subJobs []*jobsdb.JobT
+	hasMore bool
+}
+
+func jobSplitter(jobs []*jobsdb.JobT) []subJobT {
+	subJobCount := 1
+	if len(jobs)/subJobSize > 1 {
+		subJobCount = len(jobs) / subJobSize
+		if len(jobs)%subJobSize != 0 {
+			subJobCount++
+		}
+	}
+	var subJobs []subJobT
+	for i := 0; i < subJobCount; i++ {
+		if i == subJobCount-1 {
+			//all the remaining jobs are sent in last sub-job batch.
+			subJobs = append(subJobs, subJobT{
+				subJobs: jobs,
+				hasMore: false,
+			})
+			continue
+		}
+		subJobs = append(subJobs, subJobT{
+			subJobs: jobs[:subJobSize],
+			hasMore: true,
+		})
+		jobs = jobs[subJobSize:]
+	}
+
+	return subJobs
+}
+
 // mainPipeline: new way of handling jobs
 //
 // [getJobs] -chProc-> [processJobsForDest] -chTrans-> [transformations] -chStore-> [Store]
 func (proc *HandleT) mainPipeline(ctx context.Context) {
 	//waiting for reporting client setup
+	proc.logger.Info("Processor mainPipeline started")
+	proc.logger.Info("Processor subJobSize= ", subJobSize)
+	proc.logger.Info("Processor pipelineBufferedItems= ", pipelineBufferedItems)
```

Comment:
[minor] For more concise logs:

```suggestion
	proc.logger.Info("Processor mainPipeline started")
	proc.logger.Info("Processor subJobSize= ", subJobSize)
	proc.logger.Info("Processor pipelineBufferedItems= ", pipelineBufferedItems)
```